### PR TITLE
sqlc: beds, rooms, and patients

### PR DESCRIPTION
**Pull request by FoseFx**
## Which issues does this pull request close?
Progresses #563, #562,  #570

Removes beds, rooms and patients repos (now generated using sqlc) and all direct usage of the corresponding models.

BREAKING: makes GetPatientList's Task's AssignedUserID optional.

Affected by this PR:

- *CreateBed*
- *GetBed*
- *GetBedByPatient*
- *GetBeds*
- **UpdateBed**
- *DeleteBed*
- **CreatePatient**
- *GetPatient*
- *GetPatientsByWard*
- *GetPatientAssignmentByWard*
- GetRecentPatients
- **UpdatePatient**
- *AssignBed*
- *UnassignBed*
- *DischargePatient*
- *GetPatientDetails*
- *GetPatientList*
- *DeletePatient*
- *ReadmitPatient*
- **CreateRoom**
- *GetRoom*
- **UpdateRoom**
- *GetRooms*
- *GetRoomsByWard*
- **DeleteRoom**
- *GetRoomOverviewsByWard*
- **CreateTask**
- *GetTask*

Italic Endpoints have integration tests

### CODE REVIEW

**Comment by FoseFx** [comment]
Done with Phase I: All tests are passing.

File: services/task-svc/repos/room_repo.sql [hunk 1]
```diff
@@ -0,0 +1,78 @@
+-- name: CreateRoom :one
+INSERT INTO rooms (name, organization_id, ward_id) VALUES ($1, $2, $3) RETURNING id;
+
+-- name: GetRoomWithBedsById :many
```
Comment (MaxSchaefer) [inline services/task-svc/repos/room_repo.sql:4]:
```suggestion
-- name: GetRoomWithBedsById :one
```
@FoseFx

Comment (FoseFx) [reply services/task-svc/repos/room_repo.sql:4]: nope! JOIN can lead to multiple rows

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
Looks good so far.
Found one error in Endpoint `GetRecentPatients`:

```
failed to encode args[0]: unable to encode []uuid.UUID{uuid.UUID{0x6a, 0x7a, 0x8d, 0xfe, 0x5f, 0x8f, 0x4b, 0x7f, 0xa9, 0xf6, 0xd, 0xf0, 0xd2, 0xbf, 0x14, 0x23}, uuid.UUID{0xc1, 0xf7, 0x6e, 0xd5, 0xcd, 0x56, 0x48, 0xbe, 0x96, 0x5, 0x7f, 0x6b, 0x6, 0x4d, 0xcb, 0x33}, uuid.UUID{0x9b, 0xa7, 0x7, 0x3d, 0x94, 0xc6, 0x43, 0x35, 0x99, 0xd7, 0x1b, 0x10, 0xc6, 0xf8, 0x91, 0x51}, uuid.UUID{0x97, 0xf0, 0xd2, 0x87, 0x34, 0x98, 0x41, 0x4c, 0xba, 0xb9, 0xf, 0x34, 0x1b, 0x32, 0x8a, 0xe4}} into binary format for uuid (OID 2950): cannot find encode plan

```

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**